File: app/src/test/java/io/github/tobyhs/weatherweight/api/accuweather/locations/CityTest.java
package io.github.tobyhs.weatherweight.api.accuweather.locations;

import com.google.gson.Gson;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;

import org.junit.Test;

import io.github.tobyhs.weatherweight.AppModule;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class CityTest {
    @Test
    public void parseJson() throws Exception {
        try (
                InputStream stream = getClass().getResourceAsStream("/accuweather/city.json");
                Reader reader = new InputStreamReader(stream)
        ) {
            Gson gson = AppModule.provideGson();
            City city = gson.fromJson(reader, City.class);

            assertThat(city.getKey(), is("347629"));
            assertThat(city.getLocalizedName(), is("San Francisco"));
            assertThat(city.getAdministrativeArea().getId(), is("CA"));
            assertThat(city.getCountry().getId(), is("US"));
        }
    }
}

File: app/src/test/java/io/github/tobyhs/weatherweight/api/accuweather/locations/CityTest.kt
package io.github.tobyhs.weatherweight.api.accuweather.locations

import io.github.tobyhs.weatherweight.AppModule

import java.io.InputStreamReader

import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat

import org.junit.Test

class CityTest {
    @Test
    fun parseJson() {
        javaClass.getResourceAsStream("/accuweather/city.json").use { stream ->
            InputStreamReader(stream).use { reader ->
                val moshi = AppModule.provideMoshi()
                val moshiAdapter = moshi.adapter(City::class.java)
                val city = moshiAdapter.fromJson(reader.readText())!!

                assertThat(city.key, equalTo("347629"))
                assertThat(city.localizedName, equalTo("San Francisco"))
                assertThat(city.administrativeArea.id, equalTo("CA"))
                assertThat(city.country.id, equalTo("US"))
            }
        }
    }
}